paddle/phi/backends/gpu/cuda/cuda_graph.cc

```diff
@@ -308,6 +270,79 @@ void CUDAGraph::PrintToDotFiles(const std::string &dirname,
 #endif
 }
 
+std::vector<cudaGraphExecuterSetter_t>
+CUDAGraphNodeLauncher::GetParameterSettersForExecGraph(cudaGraph_t graph) {
+  size_t num_nodes;
+  PADDLE_ENFORCE_GPU_SUCCESS(cudaGraphGetNodes(graph, nullptr, &num_nodes));
+  std::vector<cudaGraphNode_t> nodes(num_nodes);
+  PADDLE_ENFORCE_GPU_SUCCESS(
+      cudaGraphGetNodes(graph, nodes.data(), &num_nodes));
+
+  std::vector<std::function<void(cudaGraphExec_t)>> hooks;
+  for (auto node : nodes) {
+    CUgraphNode cuNode = node;
+    CUgraphNodeType pType;
+    PADDLE_ENFORCE_GPU_SUCCESS(dynload::cuGraphNodeGetType(cuNode, &pType));
```

Comment:
#### The Importance of the CUDA Driver API in `CUDAGraphNodeLauncher`

Our revised implementation underscores the importance of the CUDA driver API. In the original code, we relied on the `cudaGraphKernelNodeGetParams` from the CUDA runtime API to retrieve node parameters. This approach, however, occasionally resulted in the `cudaErrorInvalidDeviceFunction` error. Kindly refer to the code [here](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/phi/backends/gpu/cuda/cuda_graph.cc#L221).

This error arises when attempting to retrieve a node from another shared library - such as CUDNN kernels or user-defined kernels. In the realm of the CUDA driver API, a shared library is represented by the `CUlibrary` handle, details of which can be found [here](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__LIBRARY.html). Contrarily, the CUDA runtime API simplifies and hides this structure from the user, primarily presuming that kernel function pointers originate from the same library. This assumption leads to the aforementioned error when accessing kernel function pointers from distinct libraries. Direct engagement with the CUDA driver API avoid this issue.

It's crucial to note that, with this change, users of the `CUDAGraphNodeLauncher` are responsible for securing the `CUFunction` structure, particularly since the kernel could reside in diverse libraries.